### PR TITLE
Add I18n translations support in views

### PR DESCRIPTION
**Pull request by anthony-robin**

### CODE REVIEW

**Comment by tatey** [comment]
Hello @anthony-robin. Thanks for taking the initiative on this. @coop or I will review it over the weekend.

**Comment by anthony-robin** [comment]
Great ! Let me know if there are some stuffs you wants me to change...

File: README.markdown [hunk 1]
```diff
@@ -89,13 +97,31 @@ are always welcome.
...
+# config/application.rb
+config.assets.enabled = true
+config.assets.precompile << 'delayed/web/application.css'
+```
 
 This step is skipped if your project is configured not to use the asset
 pipeline.
 
+## I18n
+If you want to translate flash messages, titles, sentences, ... you can set your own I18n translations to override defaults one:
```
Comment (tatey) [inline README.markdown]: Should be a blank line between headings and paragraphs.

File: README.markdown [hunk 2]
```diff
@@ -89,13 +97,31 @@ are always welcome.
...
+  delayed/web:
+    views:
+      id: ID
+      status: Status
+      created_at: Created At
+      run_at: Run At
+      attempts: Attempts
+      last_error: Last Error
+```
+You can find the full list [here](https://github.com/gemsfix/delayed-web/blob/master/config/locales/en.yml).
```
Comment (tatey) [inline README.markdown]: Blank line after code block.

File: app/views/delayed/web/jobs/index.html.erb [hunk 3]
```diff
@@ -7,13 +7,13 @@
...
-      <th>Attempts</th>
-      <th style="width: 190px;">Last Error</th>
-      <th style="width: 115px;">Actions</th>
+      <th><%= t('delayed/web.views.id') %></th>
+      <th><%= t('delayed/web.views.status') %></th>
+      <th><%= t('delayed/web.views.created_at') %></th>
+      <th><%= t('delayed/web.views.run_at') %></th>
+      <th><%= t('delayed/web.views.attempts') %></th>
+      <th style="width: 190px;"><%= t('delayed/web.views.last_error') %></th>
+      <th style="width: <%= I18n.locale == :en ? '115' : '160' %>px;"><%= t('delayed/web.views.actions') %></th>
```
Comment (tatey) [inline app/views/delayed/web/jobs/index.html.erb]: This feels fragile for other locales. Why not just make it 160px always?

Comment (anthony-robin) [inline app/views/delayed/web/jobs/index.html.erb]: The problem is that french words are larger than english, so if I set 160px for all languages, there will be many spaces between buttons in english.

**Comment by anthony-robin** [comment]
That's my first pull request. Should I edit my code and resubmit a new one ? Thanks !

File: app/views/delayed/web/jobs/index.html.erb (hunk 3 above)
Comment (tatey) [inline app/views/delayed/web/jobs/index.html.erb]: Sure, but what happens if someone adds another language and then the optimal width is 130px. The point is that we should pick the route that has the least amount of effort to maintain over time. Consider picking a middle size and then using CSS to truncate. Eg: `text-overflow`.

**Comment by tatey** [comment]
> That's my first pull request. Should I edit my code and resubmit a new one 

Do the work as a new commit and then use `git commit --fixup <sha>` where the `<sha>` is the commit where the work should have originally occurred. Once all that looks good we can rebase against master, squashing the fix ups, and we have a nice, clean history.

> Also, english is not my first language, so feel free to update README instructions if the section I added doesn't use correct syntax.

No, it was great 😄

**Comment by tatey** [comment]
@andychisholm I've updated master to run against Ruby 2.3 and Rails 5 and the specs are all 💚.

**Comment by anthony-robin** [comment]
I made the changes you asked for (I hope I didn't mess too much things with git history ^^) !
I also added an alert confirmation before to destroy a job (I forgot to do it last time).

Thanks !

**Comment by tatey** [comment]
Thanks @anthony-robin. I’ll try to have a look at this tonight.

**Comment by tatey** [comment]
@anthony-robin This is perfect 👌 

Now that I have completed the review do you mind doing a `git rebase -i master --autosquash` and it will squash the fixups and replay your commits on top of master. Then I will merge and make a release 😄

**Comment by anthony-robin** [comment]
Great ! I'v used the command `git rebase -i master --autosquash` but what should I do next ? 😕 
Should I merge to my branch `master` ?
I tried to push to `feature/i18n-views` and it says me to pull first (which give me conflicts).

I'm lost, thanks for your help ! 👍

**Comment by tatey** [comment]
> I tried to push to my feature/i18n-views fork branch and it says me to pull first (which give me conflicts). Is that normal ?

Yes, you need to force push with `git push --force origin feature/i18n-views`. `origin` is what ever you have called your remote.

The reason you need to force push is because you "rewriting" history, but that's okay in a pull request under review.

**Comment by anthony-robin** [comment]
Done ! I pushed force to `origin feature/i18n-views` as you explained.

**Comment by tatey** [comment]
Thanks for your contribution @anthony-robin 😄 

I've merged into master and released to ruby gems https://rubygems.org/gems/delayed-web/versions/0.4.2.

**Comment by anthony-robin** [comment]
Perfect ! That was a pleasure to contribute :)

**Comment by tatey** [comment]
You did good mate.

**Pull request merged**